website/content/docs/connect/gateways/ingress-gateway.mdx

```diff
@@ -68,4 +68,212 @@ must also provide `agent:read` for its node's name in order to discover the agen
 
 ~> [Configuration entries](/docs/agent/config-entries) are global in scope. A configuration entry for a gateway name applies
 across all federated Consul datacenters. If ingress gateways in different Consul datacenters need to route to different
-sets of services within their datacenter then the ingress gateways **must** be registered with different names.
+sets of services within their datacenter, then the ingress gateways **must** be registered with different names.
+
+<!-- Add a "permalink" anchor here since this title is long and may be edited
+     but we need to deep-link to it elsewhere -->
+<a name="sds"></a>
+
+## Custom TLS Certificates via Secret Discovery Service (SDS)
+
+~> **Advanced Topic** This is a low-level feature designed for developers
+building integrations with custom TLS management solutions.
+
+Consul 1.11 added support for Ingress Gateways to serve TLS certificates to
+inbound traffic that are sourced from an external service. The external service
+must implement Envoy's [gRPC Secret Discovery
+Service](https://www.envoyproxy.io/docs/envoy/latest/configuration/security/secret)
+(or SDS) API.
+
+The steps below are necessary to fully configure an Ingress Gateway with TLS
+certificates from an SDS source. They assume the operator is familiar with Envoy
+ configuration and the SDS protocol.
+
+### 1. Configure Static SDS Cluster(s).
+
+Each Ingress Gateway Envoy instance must have one or more additional [static
+clusters](/docs/connect/proxies/envoy#envoy_extra_static_clusters_json) defined
+when registering the ingress gateway to specify how to connect to the external
+service. This requires a manual registration of the Ingress Gateway proxy rather
```

Comment:
Heh no it's far from obvious - I was trying to be as precise as possible but that's clearly not been successful or at least not made it easier to understand 😄 .

Basically: an ingress gateway config is "logical" with a name, but then one or more actual Envoy instances are run to implement the gateway. I want to be specific that it's when running those Envoy's that make up this specific gateway that you need to define these extra things. It's not all Envoy instances because Gateways that are not using SDS don't need this, nor do Sidecar proxies in the mesh.

So "Each gateway instance" wasn't precise enough because it's only ingress gateways. "Each Ingress Gateway instance" is confusing because that could be refering to the logical gateway (i.e. it's ConfigEntry). We could say "Each Ingress Gateway Proxy", I avoided that just because I expect this feature to only with with Envoy and it's technically possible that someone could use a different proxy as an ingress (though very unlikely and not something we are expecting to work on any time soon).

How about this as an improvement:
```suggestion
Each Envoy proxy that makes up this Ingress Gateway must define one or more additional [static
clusters](/docs/connect/proxies/envoy#envoy_extra_static_clusters_json) when registering. These additional clusters define how Envoy should connect to the required SDS service(s). Defining extra clusters in Envoy's bootstrap configuration requires a manual registration of the Ingress Gateway with Consul proxy rather
```

Reply:
Yes, I think that's much better